src/utils/jsx.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+/* @flow */
+
+import * as t from "babel-types";
+import type { BabelNodeExpression, BabelNodeJSXMemberExpression, BabelNodeJSXIdentifier } from "babel-types";
+import invariant from "../invariant.js";
+
+export function convertExpressionToJSXIdentifier(
+  expr: BabelNodeExpression
+): BabelNodeJSXMemberExpression | BabelNodeJSXIdentifier {
+  switch (expr.type) {
+    case "ThisExpression":
+      return t.jSXIdentifier("this");
+    case "Identifier":
+      return t.jSXIdentifier(expr.name);
+    case "StringLiteral":
+      return t.jSXIdentifier(expr.value);
+    case "MemberExpression":
+      if (expr.computed) {
+        throw new Error("Cannot inline computed expressions in JSX type.");
```

Comment:
All errors should fall into one of two categories:
- user facing errors. In that case, a FatalError should be thrown (or @hermanventer probably has a good suggestion on how to do it).
- internal errors / bugs. In that case, state an `invariant`.

Reply:
I've updated it to an invariant :)